libraries/Microsoft.Bot.Connector/Authentication/ChannelValidation.cs

```diff
@@ -48,18 +48,19 @@ public static class ChannelValidation
         /// HttpClient is used for making those calls. Those calls generally require TLS connections, which are expensive to
         /// setup and teardown, so a shared HttpClient is recommended.</param>
         /// <param name="channelId">The ID of the channel to validate.</param>
+        /// <param name="authConfig">The optional authentication configuration.</param>
         /// <returns>
         /// A valid ClaimsIdentity.
         /// </returns>
-        public static async Task<ClaimsIdentity> AuthenticateChannelToken(string authHeader, ICredentialProvider credentials, HttpClient httpClient, string channelId)
+        public static async Task<ClaimsIdentity> AuthenticateChannelToken(string authHeader, ICredentialProvider credentials, HttpClient httpClient, string channelId, AuthenticationConfiguration authConfig = null)
         {
             var tokenExtractor = new JwtTokenExtractor(
                 httpClient,
                 ToBotFromChannelTokenValidationParameters,
                 OpenIdMetadataUrl,
                 AuthenticationConstants.AllowedSigningAlgorithms);
 
-            var identity = await tokenExtractor.GetIdentityAsync(authHeader, channelId);
+            var identity = await tokenExtractor.GetIdentityAsync(authHeader, channelId, authConfig?.RequiredEndorsements);
             if (identity == null)
```

Comment:
Can we eliminate the null programming? Just have the default RequiredEndorsements be an empty List<string> or similar. 

Null programming always ends badly...

Reply:
Sure! I like it, less code :)